smartsim/_core/mli/infrastructure/control/workermanager.py

```diff
@@ -245,13 +243,23 @@ def _on_iteration(self) -> None:
             return
 
         timings = []  # timing
-        # perform default deserialization of the message envelope
-        request_bytes: bytes = self._task_queue.recv()
+
+        bytes_list: t.List[bytes] = self._task_queue.recv()
+        request_bytes: bytes = b""
+        tensor_list = []
+
+        if bytes_list:
+            request_bytes = bytes_list[0]
+            tensor_list = bytes_list[1:]
 
         interm = time.perf_counter()  # timing
         request = deserialize_message(
             request_bytes, self._comm_channel_type, self._device
         )
+
+        if request.input_meta and tensor_list:
```

Comment:
Maybe The logic from 248 to 264 (and the deserialization_message() content) would be better encapsulated in a ``unpack_request``.  I think _on_iteration should have minimal manipulation of the request based on the serialiation and communication specifics.

Reply:
(Does this make it more difficult to do perf timing though?)

Reply:
I completely agree, but maybe we wait to refactor `_on_iteration` until we're solid with performance timing? It might make it more difficult.